grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptData.java

```diff
@@ -82,14 +90,17 @@ private void handleConcept(Representation halResource, Concept concept, int sepa
 
         generateStateAndLinks(halResource, concept);
 
-        if (embedType && concept.type() != null &&
-                (typesInQuery.contains(concept.type().getName())
-                        || (concept.type().superType() != null &&
-                        typesInQuery.contains(concept.type().superType().getName()))))
-            embedType(halResource, concept);
+        if (embedType && concept.isInstance()) {
+            Instance instance = concept.asInstance();
+            if (typesInQuery.contains(instance.type().getName())
+                    || (instance.type().superType() != null &&
+                    typesInQuery.contains(instance.type().superType().getName())))
+                embedType(halResource, instance);
+        }
 
-        if (concept.type() == null)
-            embedType(halResource, concept);
+        if (concept.isInstance()) {
+            embedType(halResource, concept.asInstance());
+        }
```

Comment:
I think I messed up here, seems to be backwards to how it should be (previously it was: only embed type if there... isn't a type)